Library/Homebrew/rubocops/checksum_cop.rb

```diff
@@ -0,0 +1,61 @@
+require_relative "./extend/formula_cop"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      class Checksum < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          %w[stable devel head].each do |name|
+            next unless spec_node = find_block(body_node, name.to_sym)
```

Comment:
Looks like this will`next` without an explicit `stable do` block whereas we want to ensure a root `sha256` is checked in that case. I think having a separate section of this method for `stable` outside of `stable do` would make sense; you can then put `find_blocks(body_node, :resource)` in there too to avoid the `if` below.